src/coord/src/coord.rs

```diff
@@ -3354,22 +3354,31 @@ impl Coordinator {
 
         let since = self.least_valid_read(&id_bundle, compute_instance);
 
+        let mut advance_to_timestamp: Option<Timestamp> = None;
+        let mut advance_to_since = false;
+        let mut advance_to_upper = false;
+        let mut advance_to_table_ts = false;
+        let mut advance_to_linearizability_ts = false;
+        let mut bump_linearizability_ts = false;
```

Comment:
I read this and immediately concluded "this cannot be the best way to write this" not yet knowing what you will write. :D Let me loop back around after I read the implementation.

Related: Having comments atop the block of code would prep expectations about what it does or should do, and help the readers and reviewer!

Reply:
I read it all and do not understand the implementation!

Reply:
The "bump" variable here was based on the assumption that we would implement linearizability with a timeline timestamp that queries could advance and other queries would use when determining timestamps. I think that approach is not how we are going to handle things, so this variable probably goes away.